lib/ansible/modules/source_control/git.py

```diff
@@ -447,6 +469,17 @@ def clone(git_path, module, repo, dest, remote, depth, version, bare,
     if verify_commit:
         verify_commit_sign(git_path, module, dest, version)
 
+    if separate_git_dir_fallback:
+        separate_git_dir = os.path.abspath(separate_git_dir)
+        fallback_cmd = ['mv', os.path.join(dest, '.git'), separate_git_dir]
```

Comment:
What will happen if user specifies non-existent directory?
What will happen if user specifies a directory inside of the workdir?

Reply:
It works well when dir does not exist and dir inside of workdir.
But when separate-git-dir already exists, it will fail. I will fix this.